rmc/utils/constraint.py

```diff
@@ -1454,63 +1548,110 @@ def fix_region_start_stop_aas(
+    cds_ht = transcript_cds.ht()
+    cds_ht = cds_ht.filter(missing_transcripts.contains(cds_ht.transcript))
+    cds_ht = cds_ht.annotate(
+        exon_start=cds_ht.interval.start,
+        exon_end=cds_ht.interval.end,
+    )
+
+    # Create two versions of CDS HT to fix missing region start and stop AAs
+    def _create_exon_position_ht(
+        cds_ht: hl.Table, fix_missing_start: bool, overwrite_temp: bool
+    ) -> hl.Table:
+        """
+        Create Table keyed by either exon start or stop position.
+
+        :param cds_ht: CDS HT resource filtered to transcripts missing region start or stop
+            AA annotations only.
+        :param fix_missing_start: Whether the output HT is designed to fix region starts
+            that are missing AA annotations.
+        :param overwrite_temp: Whether to overwrite temporary data.
+            If False, will read existing temp data rather than overwriting.
+            If True, will overwrite temp data.
+        :return: CDS HT keyed by either exon start or stop position and transcript.
+        """
+        if fix_missing_start:
+            # Region starts missing AAs need to get adjusted to use the AA from the previous
+            # exon start
+            # Reorder HT so that _prev_nonnull returns the next exon start
+            checkpoint_path = f"{TEMP_PATH_WITH_FAST_DEL}/cds_start_fix.ht"
+            cds_ht = cds_ht.order_by(
+                hl.asc(cds_ht.transcript), hl.desc(cds_ht.exon_start)
+            )
+            cds_ht = cds_ht.annotate(
+                next_exon_start=hl.scan._prev_nonnull(cds_ht.exon_start)
+            )
+            cds_ht = cds_ht.key_by(cds_ht.exon_stop, cds_ht.transcript)
+        else:
+            checkpoint_path = f"{TEMP_PATH_WITH_FAST_DEL}/cds_stop_fix.ht"
+            cds_ht = cds_ht.annotate(
+                prev_exon_stop=hl.scan._prev_nonnull(cds_ht.exon_stop)
+            )
+            cds_ht = cds_ht.key_by(cds_ht.exon_start, cds_ht.transcript)
+
+        cds_ht = cds_ht.checkpoint(
+            checkpoint_path,
+            _read_if_exists=not overwrite_temp,
+            overwrite=overwrite_temp,
+        )
+        return cds_ht
+
+    # Get relevant exon start and stop positions
+    start_fix_ht = _create_exon_position_ht(
+        cds_ht=cds_ht, fix_missing_start=True, overwrite_temp=overwrite_temp
     )
-    # Adjust start/stop coordinates to closest missense positions
-    missing_ht = missing_ht.transmute(
-        start_coordinate=hl.if_else(
-            hl.is_defined(missing_ht.start_aa),
-            hl.locus(missing_ht.start_coordinate.contig, missing_ht.closest_start_mis),
-            missing_ht.start_coordinate,
+    stop_fix_ht = _create_exon_position_ht(
+        cds_ht=cds_ht, fix_missing_start=False, overwrite_temp=overwrite_temp
+    )
+    missing_ht = missing_ht.annotate(
+        next_exon_start=hl.or_missing(
+            hl.is_missing(missing_ht.start_aa),
+            stop_fix_ht[
```

Comment:
I think these are switched
```suggestion
            start_fix_ht[
```

Reply:
BAHA I always miss copy paste errors, thank you for catching